go/pkg/cas/upload.go

```diff
@@ -68,8 +70,27 @@ func (c *Client) Upload(ctx context.Context, inputC <-chan *UploadInput) (stats
 		fsSem:   semaphore.NewWeighted(int64(c.FSConcurrency)),
 	}
 
+	// Initialize checkBundler, which checks if a blob is present on the server.
+	u.checkBundler = bundler.NewBundler(&uploadItem{}, func(items interface{}) {
+		// Handle errors and context cancelation via errgroup.
+		eg.Go(func() error {
```

Comment:
yes, the goal is stop everything when error happens. Also stop checking if another goroutine exited.

yeah, handling two contexts sounds fairly complicated. I am not sure how that would work.

I don't think we'd need two contexts. We can make a cancel func before creating errgroup, but then it would be more code, and it would be easier to breed a bug.

btw, it doesn't spin a new thread (at least in "OS thread" sense). Goroutines are pretty lightweight and Golang encourages to create them when needed https://screenshot.googleplex.com/5BcdW6eFhWkMYqs
We shouldn't shy away from creating goroutines when it helps.